packages/grafana-ui/src/components/ConfirmModal/ConfirmModal.tsx

```diff
@@ -1,50 +1,86 @@
-import React, { FC, useContext } from 'react';
+import React, { useState } from 'react';
 import { css } from '@emotion/css';
 import { Modal } from '../Modal/Modal';
 import { IconName } from '../../types/icon';
 import { Button } from '../Button';
-import { stylesFactory, ThemeContext } from '../../themes';
+import { useStyles } from '../../themes';
 import { GrafanaTheme } from '@grafana/data';
-import { HorizontalGroup } from '..';
+import { HorizontalGroup, Input } from '..';
+import { selectors } from '@grafana/e2e-selectors';
 
-export interface Props {
+export interface ConfirmModalProps {
   /** Toggle modal's open/closed state */
   isOpen: boolean;
   /** Title for the modal header */
   title: string;
   /** Modal content */
   body: React.ReactNode;
+  /** Modal description */
+  description?: React.ReactNode;
   /** Text for confirm button */
   confirmText: string;
   /** Text for dismiss button */
   dismissText?: string;
   /** Icon for the modal header */
   icon?: IconName;
+  /** Text user needs to fill in before confirming */
+  confirmationText?: string;
+  /** Text for alternative button */
+  alternativeText?: string;
   /** Confirm action callback */
   onConfirm(): void;
   /** Dismiss action callback */
   onDismiss(): void;
+  /** Alternative action callback */
+  onAlternative?(): void;
 }
 
-export const ConfirmModal: FC<Props> = ({
+export const ConfirmModal = ({
   isOpen,
   title,
   body,
+  description,
   confirmText,
+  confirmationText,
   dismissText = 'Cancel',
+  alternativeText,
   icon = 'exclamation-triangle',
   onConfirm,
   onDismiss,
-}) => {
-  const theme = useContext(ThemeContext);
-  const styles = getStyles(theme);
+  onAlternative,
+}: ConfirmModalProps): JSX.Element => {
+  const [disabled, setDisabled] = useState<boolean>(Boolean(confirmationText));
```

Comment:
I think `<boolean>` shouldn't be needed here

Reply:
Fixed!